File: frontend/src/api/workflowGrowth.ts
import { ApiClient } from './request'

export enum GrowthMemoryStatus {
  PENDING = 'pending',
  APPROVED = 'approved',
  REJECTED = 'rejected',
  EXPIRED = 'expired'
}

export enum GrowthSuggestionStatus {
  PENDING = 'pending',
  ACCEPTED = 'accepted',
  DISMISSED = 'dismissed',
  EXPIRED = 'expired'
}

export enum GrowthMemoryScope {
  TASK_TEMP = 'task_temp',
  OBJECT_TRACKING = 'object_tracking',
  USER_PREFERENCE = 'user_preference',
  RESEARCH_ASSET = 'research_asset',
  PATTERN = 'pattern'
}

export interface GrowthEvidenceRef {
  type: string
  id: string
  label?: string
}

export interface GrowthMemoryItem {
  memory_id: string
  user_id: string
  scope: GrowthMemoryScope | string
  status: GrowthMemoryStatus | string
  review_level: string
  source_type: string
  source_id: string
  task_type?: string
  workflow_id?: string
  object_type?: string
  object_key?: string
  title: string
  summary: string
  content: string
  structured_payload?: Record<string, any>
  confidence: number
  evidence_refs: GrowthEvidenceRef[]
  created_at: string
  updated_at: string
}

export interface GrowthSuggestion {
  suggestion_id: string
  user_id: string
  suggestion_type: string
  status: GrowthSuggestionStatus | string
  source_type: string
  source_id: string
  task_type?: string
  workflow_id?: string
  object_type?: string
  object_key?: string
  title: string
  summary: string
  suggested_action?: Record<string, any>
  confidence: number
  source_refs: GrowthEvidenceRef[]
  activation_result?: {
    status?: string
    target?: string
    mode?: string
    session_id?: string
    ai_message?: string
    error?: string
  }
  created_at: string
  updated_at: string
}

export interface TaskGrowthContextPayload {
  task_id: string
  memory_items: GrowthMemoryItem[]
  suggestions: GrowthSuggestion[]
  summary: {
    memory_items_count: number
    suggestions_count: number
  }
}

export function getTaskGrowthContext(taskId: string) {
  return ApiClient.get<TaskGrowthContextPayload>(`/api/workflow-growth/tasks/${taskId}`)
}

export function approveGrowthMemory(memoryId: string) {
  return ApiClient.post<{ success: boolean; message: string }>(`/api/workflow-growth/memories/${memoryId}/approve`)
}

export function rejectGrowthMemory(memoryId: string) {
  return ApiClient.post<{ success: boolean; message: string }>(`/api/workflow-growth/memories/${memoryId}/reject`)
}

export function acceptGrowthSuggestion(suggestionId: string) {
  return ApiClient.post<GrowthSuggestion>(`/api/workflow-growth/suggestions/${suggestionId}/accept`)
}

export function dismissGrowthSuggestion(suggestionId: string) {
  return ApiClient.post<{ success: boolean; message: string }>(`/api/workflow-growth/suggestions/${suggestionId}/dismiss`)
}

export const GrowthMemoryStatusNames: Record<string, string> = {
  pending: '待审核',
  approved: '已批准',
  rejected: '已拒绝',
  expired: '已过期'
}

export const GrowthSuggestionStatusNames: Record<string, string> = {
  pending: '待处理',
  accepted: '已接受',
  dismissed: '已忽略',
  expired: '已过期'
}

export const GrowthMemoryScopeNames: Record<string, string> = {
  task_temp: '任务临时记忆',
  object_tracking: '对象跟踪记忆',
  user_preference: '用户偏好',
  research_asset: '研究资产',
  pattern: '经验模式'
}

File: app/routers/workflow_growth.py
"""
工作流成长 API
"""

from typing import Optional

from fastapi import APIRouter, Depends, HTTPException, Query

from app.models.agent_growth import GrowthMemoryScope, GrowthMemoryStatus, GrowthSuggestionStatus
from app.routers.auth_db import get_current_user
from app.services.workflow_growth_service import get_workflow_growth_service

router = APIRouter(prefix="/api/workflow-growth", tags=["workflow-growth"])


@router.get("/memories", summary="获取成长记忆列表")
async def list_growth_memories(
    status: Optional[GrowthMemoryStatus] = Query(None, description="状态过滤"),
    scope: Optional[GrowthMemoryScope] = Query(None, description="范围过滤"),
    task_id: Optional[str] = Query(None, description="按任务过滤"),
    limit: int = Query(50, ge=1, le=200, description="返回数量"),
    user: dict = Depends(get_current_user),
):
    service = get_workflow_growth_service()
    items = await service.list_memory_items(
        user_id=str(user["id"]),
        status=status,
        scope=scope,
        task_id=task_id,
        limit=limit,
    )
    return {"success": True, "data": {"items": items, "total": len(items)}, "message": "ok"}


@router.get("/memories/pending", summary="获取待审核成长记忆")
async def list_pending_growth_memories(
    limit: int = Query(50, ge=1, le=200, description="返回数量"),
    user: dict = Depends(get_current_user),
):
    service = get_workflow_growth_service()
    items = await service.list_memory_items(
        user_id=str(user["id"]),
        status=GrowthMemoryStatus.PENDING,
        limit=limit,
    )
    return {"success": True, "data": {"items": items, "total": len(items)}, "message": "ok"}


@router.post("/memories/{memory_id}/approve", summary="批准成长记忆")
async def approve_growth_memory(memory_id: str, user: dict = Depends(get_current_user)):
    service = get_workflow_growth_service()
    success = await service.approve_memory_item(str(user["id"]), memory_id)
    if not success:
        raise HTTPException(status_code=404, detail="成长记忆不存在")
    return {"success": True, "message": "已批准成长记忆"}


@router.post("/memories/{memory_id}/reject", summary="拒绝成长记忆")
async def reject_growth_memory(memory_id: str, user: dict = Depends(get_current_user)):
    service = get_workflow_growth_service()
    success = await service.reject_memory_item(str(user["id"]), memory_id)
    if not success:
        raise HTTPException(status_code=404, detail="成长记忆不存在")
    return {"success": True, "message": "已拒绝成长记忆"}


@router.get("/suggestions", summary="获取成长建议列表")
async def list_growth_suggestions(
    status: Optional[GrowthSuggestionStatus] = Query(None, description="状态过滤"),
    task_id: Optional[str] = Query(None, description="按任务过滤"),
    limit: int = Query(50, ge=1, le=200, description="返回数量"),
    user: dict = Depends(get_current_user),
):
    service = get_workflow_growth_service()
    items = await service.list_suggestions(
        user_id=str(user["id"]),
        status=status,
        task_id=task_id,
        limit=limit,
    )
    return {"success": True, "data": {"items": items, "total": len(items)}, "message": "ok"}


@router.post("/suggestions/{suggestion_id}/accept", summary="接受成长建议")
async def accept_growth_suggestion(suggestion_id: str, user: dict = Depends(get_current_user)):
    service = get_workflow_growth_service()
    result = await service.accept_suggestion(str(user["id"]), suggestion_id)
    if not result:
        raise HTTPException(status_code=404, detail="成长建议不存在")
    return {"success": True, "data": result, "message": "已接受成长建议"}


@router.post("/suggestions/{suggestion_id}/dismiss", summary="忽略成长建议")
async def dismiss_growth_suggestion(suggestion_id: str, user: dict = Depends(get_current_user)):
    service = get_workflow_growth_service()
    success = await service.dismiss_suggestion(str(user["id"]), suggestion_id)
    if not success:
        raise HTTPException(status_code=404, detail="成长建议不存在")
    return {"success": True, "message": "已忽略成长建议"}


@router.get("/tasks/{task_id}", summary="获取任务的成长提取结果")
async def get_task_growth_context(task_id: str, user: dict = Depends(get_current_user)):
    service = get_workflow_growth_service()
    data = await service.get_task_context(str(user["id"]), task_id)
    return {"success": True, "data": data, "message": "ok"}